lib/charms/layer/basic.py

```diff
@@ -278,6 +284,29 @@ def _load_wheelhouse_versions():
     return versions
 
 
+def _add_back_versions(pkgs, _versions):
+    """Add back the version strings to each of the packages.
+
+    The versions are LooseVersion() from _load_wheelhouse_versions().  This
+    function strips the ".zip" or ".tar.gz" from the end of the version string
+    and adds it back to the package in the form of <package_name>==<version>
+    """
+    def _strip_ext(s):
+        _s = str(s)
+        for ending in [".zip", ".tar.gz"]:
+            if _s.endswith(ending):
+                return _s[:-len(ending)]
+        return _s
+
+    def _maybe_make_version(k):
```

Comment:
I don't understand why this is called "maybe"? Shouldn't it be called "add_version_remove_extension" or "enforce_with_version_without_extension" or something like that?

Reply:
It's called `_maybe*` as it might make the key `k` into a versioned `package==version`.  However, I could probably do with better names.